### DISCUSSION TITLE
unable to deploy to google cloud run

### DISCUSSION THREAD

**Discussion by rockycamacho** [serverpod/serverpod#883] [Q&A] [answered]
Hello I tried running the cloud-run-deploy.sh script and received an error during "Creating Revision"

`Failed to load passwords. Serverpod cannot not start.`

Please help.

**Comment by vlidholt** [comment]
Did you clone your repository from somewhere else or running the deployment from CI?

The `config/passwords.yaml` file is not version controlled for security reasons.

Also, you may want to verify that you have a `.gcloudignore` file in the root folder of your project. If the `.gcloudignore` file is missing, cloud run will use the `.gitignore` file instead (which will ignore the `passwords.yaml` file.

**Reply by rockycamacho** [reply, marked as answer]
Thanks @vlidholt, adding the `.gcloudignore` file worked for me.

**Reply by rockycamacho** [reply]
I later encountered a different problem with the program running with default arguments (mode=development, role=monolith). I solved it by copying the `server.dart` to `bin/server.dart` and adding --mode, staging, --role and serverless to the program arguments. after that it was able to deploy properly. Is there a more correct solution to the arguments not being passed properly?

**Reply by vlidholt** [reply]
It sounds like you perhaps have upgraded from an earlier version of Serverpod? These should be preconfigured when you create a new project and the cloudignore file should be in place.

**Reply by rockycamacho** [reply]
is there a list of upgrade notes somewhere? I cant find any on the documentation site

**Reply by vlidholt** [reply]
Sorry for the late reply here. Upgrading is pretty straight forward. Just make sure the cli is updated.

`dart pub global activate serverpod_cli`

Then make sure all your references to serverpod is pointing to the correct version in the server/client/flutter packages.

For the deployment features it is slightly more complicated as the scripts get added when you create a new project. It may be easiest to create a new project with the same name and copy over any script files (which seems like what you've done).
